### PR TITLE
Fix hmm fit shape check

### LINKED ISSUE #1018: [BUG] Check for prior shape in HMM.fit is wrong
**Describe the bug**
For the HMM.fit method priors (if I understood correctly), should have the shape `(n, length, n_states)`, if the data has the shape `(n, length, dim)`.
However, in `partition_sequences` (https://github.com/jmschrei/pomegranate/blob/c77f967a2b66505b42a4fc4063fcf1d26406a9a5/pomegranate/_utils.py#L410), the code checks that prior has the exact same shape as `X`.
I think this is incorrect.
The same bug will occure for other input shapes to fit as well.

**To Reproduce**
To reproduce, we can simple change the last line of the semi-supervised example to call `fit` instead of `predict_proba`:

```python
from pomegranate.hmm import DenseHMM
from pomegrante.distributions import Normal

d1 = Normal([1.0], [1.0], covariance_type='diag')
d2 = Normal([3.0], [1.0], covariance_type='diag')

model = DenseHMM([d1, d2], [[0.7, 0.3], [0.3, 0.7]], starts=[0.4, 0.6])

priors = torch.ones(1, 10, 2) / 2
priors[0, 5, 0], priors[0, 5, 1] = 0, 1
priors

# The following line is changed compared to the example
model.fit(X, priors=priors)
# -> Raises Value Error
```

### PR DESCRIPTION
**Pull request by AKuederle**
This is a proposal on how to fix #1018 .

It basically forwards the number of states to `partition_sequence` to check the correct shape for the priors.

So far I only tested that this will not raise an error. I did not check, if the priors are used correctly in training.

Note: This also includes the fix proposed in #1017. If this PR is merged #1017 can be closed. If #1017 is merged separably, this PR needs to be rebased

### CODE REVIEW

**Comment by jmschrei** [comment]
Nice! Very thorough PR. Happy to merge this, but would you mind changing the variable name from `priors_last_dim` to `n_dists`? That would be semantically more helpful.

**Comment by AKuederle** [comment]
> Nice! Very thorough PR. Happy to merge this, but would you mind changing the variable name from `priors_last_dim` to `n_dists`? That would be semantically more helpful.

Done :)

**Comment by jmschrei** [comment]
Thanks!
